src/App.tsx

```diff
@@ -8,7 +7,73 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
+import { getTodos } from './api';
+
+function isIncludesQuery(todoTitle: string, query: string) {
+  return todoTitle.toLowerCase().includes(query.toLocaleLowerCase());
+}
+
 export const App: React.FC = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[] | null>(null);
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectStatus, setSelectStatus] = useState<string>('all');
+  const [query, setQuery] = useState<string>('');
+
+  const handleSelectTodo = (todoId: number | null) => (
+    todoId
+      ? (setSelectedTodo(todos?.find(todo => todo.id === todoId) || null))
+      : setSelectedTodo(null)
+  );
+
+  const handleSelectStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectStatus(event.target.value);
+  };
+
+  const handleQueryChange = (value: string) => {
+    setQuery(value);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setInitialTodos(todosFromServer);
+      });
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code